lib/storage/role.go

```diff
@@ -287,7 +287,9 @@ func (*roleMarshaler) UnmarshalRole(data []byte) (teleservices.Role, error) {
 			return nil, trace.BadParameter(err.Error())
 		}
 		// we are ignoring error from this function on purpose here
-		role.CheckAndSetDefaults()
+		if err := role.CheckAndSetDefaults(); err != nil {
```

Comment:
How about the comment right above? :)

> we are ignoring error from this function on purpose here

I would keep all `CheckAndSetDefaults()` calls as it is everywhere. Otherwise we'll start getting errors when creating/retrieving resources from the database.

Reply:
Do we want to have validation logic that is wrong?

Reply:
Thing is, these functions are called when retrieving resources from the database which means if there's a resource that doesn't pass this check for whatever reason, the user will start getting errors in unexpected places -> more support load. This has actually happened several times in the past, that's why I remember, and I'm sure we still have clusters out with such resources. I think it was an unfortunate decision to do CheckAndSetDefaults on the unmarshal in general - the resource should be rejected in the first place if it's invalid, which I think we do now.